internal/logic/logic.go

```diff
@@ -49,5 +63,167 @@ func (l htmlPdfServiceLogic) Ping(req *model.PingRequest) *respModel.Response {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrReadFileFail),
+			Data:    nil,
+		}
+	}
+	jb, err := l.htSvc.PDFPreparer(fileBytes)
+	if err != nil {
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileConversionFail),
+			Data:    nil,
+		}
+	}
+
+	err = l.dsSvc.Set(id, jb, 0)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileStoreFail),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusCreated,
+		Message: "SUCCESS",
+		Data: map[string]interface{}{
+			"id": id,
+		},
+	}
+}
+
+func (l htmlPdfServiceLogic) HtmlToPdf(w io.Writer, req *model.GenerateReq) *respModel.Response {
+	var z map[string]interface{}
+	b, err := l.dsSvc.Get(req.Id)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFetchingFile),
+			Data:    nil,
+		}
+	}
+	err = json.NewDecoder(bytes.NewBuffer(b)).Decode(&z)
+	if err != nil {
+		log.Error("error unmarshaling JSON:" + err.Error())
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrFileParseFail),
+			Data:    nil,
+		}
+	}
+	for i, p := range z["Pages"].([]interface{}) {
+		page := p.(map[string]interface{})
+		buf, err := base64.StdEncoding.DecodeString(page["Base64PageData"].(string))
+		if err != nil {
+			log.Error("error decoding base 64 input on page" + fmt.Sprint(i) + err.Error())
+			return &respModel.Response{
+				Status:  http.StatusInternalServerError,
+				Message: codes.GetErr(codes.ErrReadFileFail),
+				Data:    nil,
+			}
+		}
+		t, err := template.New(req.Id).Parse(string(buf))
+		if err != nil {
+			log.Error(err)
+			return &respModel.Response{
+				Status:  http.StatusInternalServerError,
+				Message: codes.GetErr(codes.ErrFileParseFail),
+				Data:    nil,
+			}
+		}
+		buffer := bytes.NewBuffer(nil)
+		err = t.Execute(buffer, req.Values)
+		if err != nil {
+			log.Error(err)
+			return &respModel.Response{
+				Status:  http.StatusInternalServerError,
+				Message: codes.GetErr(codes.ErrFileStoreFail),
+				Data:    nil,
+			}
+		}
+		page["Base64PageData"] = base64.StdEncoding.EncodeToString(buffer.Bytes())
+	}
+	buff := bytes.NewBuffer(nil)
+	err = json.NewEncoder(buff).Encode(z)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrEncodingFile),
+			Data:    nil,
+		}
+	}
+	err = l.htSvc.PDFGenerator(w, buff.Bytes())
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrConvertingToPdf),
+			Data:    nil,
+		}
+	}
+	return &respModel.Response{
+		Status:  http.StatusCreated,
```

Comment:
Is this as per the spec? Moreover, this function sends the file in the response.  Don't think this is anyway required.